### PR TITLE
[Improve](complex-type) update for array/map element_at with nested complex type with local tvf

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes

Issue Number: close #xxx
improve array/map element_at with nested complex type with local tvf

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.2 seconds
 stream load tsv:          514 seconds loaded 74807831229 Bytes, about 138 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17162304274 Bytes

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run clickbench

**Comment by amorynan** [comment]
run arm

**Comment by amorynan** [comment]
run clickbench

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.29 seconds
 stream load tsv:          511 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17161972336 Bytes

**Comment by amorynan** [comment]
run arm

File: be/src/vec/functions/array/function_array_element.h [hunk 1]
```diff
@@ -356,6 +400,15 @@ class FunctionArrayElement : public IFunction {
         } else if (check_column<ColumnString>(*nested_column)) {
             res = _execute_string(offsets, *nested_column, src_null_map, *idx_col, nested_null_map,
                                   dst_null_map);
+        } else if (check_column<ColumnArray>(*nested_column)) {
+            res = _get_mapped_value<ColumnArray>(offsets, *nested_column, src_null_map, *idx_col,
```
Comment (xiaokang) [inline be/src/vec/functions/array/function_array_element.h:404]: 1. change name to _execute_common
2. impl to type awareless

Comment (amorynan) [reply be/src/vec/functions/array/function_array_element.h]: done!

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run clickbench

**Comment by amorynan** [comment]
run clickbench

**Review by xiaokang — APPROVED** [review]
LGTM

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.81 seconds
 stream load tsv:          550 seconds loaded 74807831229 Bytes, about 129 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162105499 Bytes

**Review by eldenmoon — APPROVED** [review]
LGTM

**Pull request merged**